pkg/broker/broker.go

```diff
@@ -619,6 +621,16 @@ func (a AnsibleBroker) Provision(instanceUUID uuid.UUID, req *ProvisionRequest,
 	//
 	// if err is not nil, we will just bubble that up
 
+	alreadyInProgress, err := a.isProvisionInProgress(serviceInstance)
```

Comment:
I think this call needs to be within the `if uuid.Equal....` block. 

Reason being if it is a duplicate we want to return duplicate and not give the job token to the other service instance